src/test/groovy/BusinessScenariosTests.groovy

```diff
@@ -0,0 +1,162 @@
+
+/**
+ * Created by przemek on 06.10.2016.
+ */
+
+@Slf4j
+@Requires({ System.getProperty("integrationTests").equals("true") })
+class BusinessScenariosTests extends Specification {
+
+    @Shared
+    def client
+
+    @Shared
+    User userSalesRep, userAccManager
+
+    @Shared
+    Contact createdContact
+
+    @Shared
+    String personContactName = 'person assigned to a manager'
+
+    def setupSpec() {
+        client = new Client(new Configuration.Builder()
+                .accessToken(System.getProperty("BASECRM_ACCESS_TOKEN"))
+                .build())
+
+        userSalesRep = client
+                .users()
+                .get(convertUserId(System.getProperty("userSalesRep")))
+
+        userAccManager = client
+                .users()
+                .get(convertUserId(System.getProperty("userAccManager")))
+
+        def testContactName = 'Foo Contact'
+
+        assert client
+                .contacts()
+                .list(new ContactsService.SearchCriteria().name(testContactName))
+                .size() == 0
+
+        createdContact = client.contacts().create(new Contact(name: testContactName, ownerId: userSalesRep.id, isOrganization: true))
+        await().until {
+            client
+                    .contacts()
+                    .list(new ContactsService.SearchCriteria().name(createdContact.name))
+                    .size() == 1
+        }
+    }
+
+    def convertUserId(String userId) {
+        assert userId
+        return new Long(userId)
+    }
+
+    def cleanupSpec() {
+        client
+                .deals()
+                .list(new DealsService.SearchCriteria().contactId((Long) createdContact.id))*.id
+                .each {
+            id ->
+                println(id)
+                client.deals().delete(id)
+        }
+
+        await().until {
+            client
+                    .deals()
+                    .list(new DealsService.SearchCriteria().contactId((Long) createdContact.id))
+                    .size() == 0
+        }
+
+        client
+                .contacts()
+                .delete(createdContact.id)
+
+
+        Contact tempContact = client
+                .contacts()
+                .list(new ContactsService.SearchCriteria().name((String) personContactName))[0]
+
+        if (null != tempContact) {
+            client
+                    .contacts()
+                    .delete(tempContact.id)
+        }
+    }
+
+    def getDealsStageCategory(long dealStageId, boolean active) {
+        Optional<Stage> result = client.stages()
+                .list(new StagesService.SearchCriteria().active(active))
+                .stream()
+                .filter { stage -> stage.getId().equals(dealStageId) }
+                .findAny();
+
+        assert result.present
+        result.get().category
+    }
+
+    def "new deal in incoming stage should be created automatically when new contact organization is assigned to a sales rep"() {
```

Comment:
you don't seem to have any action triggering workflow to run -> it should be here in test case, not in setupSpec